src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -257,133 +263,199 @@ private void enqueueNotification(TripMonitorNotification ...tripMonitorNotificat
+                LOG.info("Trip not checked in at least an hour. Checking.");
                 return false;
             }
         } else {
             // It's less than an hour until the trip time, start more frequent trip checks (about every 15 minutes).
             if (minutesSinceLastCheck >= 15) {
                 // Last check was more than 15 minutes ago. Check. (approx. 4 checks per hour).
                 // TODO: Change log level.
-                LOG.info("Trip {} happening soon. Checking.", trip.id);
+                LOG.info("Trip happening soon. Checking.");
                 return false;
             }
-            // If the minutes until the trip is within the lead time, check the trip every minute
-            // (assuming the loop runs every minute).
-            // TODO: Should lead time just be used for notification time limit?
-            if (minutesUntilTrip <= trip.leadTimeInMinutes) {
+            // If the trip starts within 30 minutes, check the trip every minute (assuming the loop runs every minute).
+            if (minutesUntilTrip <= 30) {
                 // TODO: Change log level.
-                LOG.info(
-                    "Trip {} happening within user's lead time ({} minutes). Checking every minute.",
-                    trip.id,
-                    trip.leadTimeInMinutes
-                );
+                LOG.info("Trip happening within 30 minutes. Checking every minute.");
                 return false;
             }
         }
         // TODO: Check that journey state is not flagged
         // TODO: Check last notification time.
         // Default to skipping check.
         // TODO: Change log level.
-        LOG.info("Trip {} criteria not met to check. Skipping.", trip.id);
+        LOG.info("Trip criteria not met to check. Skipping.");
         return true;
     }
 
     /**
-     * Generate the appropriate OTP query params for the trip for the current check. This currently involves replacing
+     * Calculate the next possible time that the itinerary is possible, advancing the targetZonedDateTime and setting
+     * the targetZonedDateTime, targetDate, otpResponse and matchingItinerary. Returns false if an error occurs or a
+     * matching itinerary isn't found on the next possible date the trip should be monitored on.
+     */
+    private boolean calculateNextItinerary() {
+        // if the trip is not active on the current zoned date time, advance until a day is found when the trip is
+        // active. It is guaranteed that the trip will be active on a certain date because a call to
+        // {@link MonitoredTrip#isInactive} should already have been made in the shouldSkipMonitoredTripCheck method.
+        while (!trip.isActiveOnDate(targetZonedDateTime)) {
+            targetZonedDateTime = targetZonedDateTime.plusDays(1);
+        }
+        targetDate = targetZonedDateTime.format(DATE_FORMATTER);
```

Comment:
Would it be more appropriate to just make targetDate a `LocalDate` type instead of a `String`? It seems like there are a lot of places where we're converting to String, but it might be simpler to read/fewer lines to just handle conversion to String in `generateTripPlanQueryParams` (which I think is the only spot it's needed as a string).

Reply:
The `targetDate` is also persisted via the JourneyState. I'm not sure I'm seeing what you're seeing as this appears to be the only place that there is a conversion from a ZonedDateTime to a string in this file.

Reply:
I guess I see your point when getting the date from a string. Another thing I'm concerned about is how a LocalDate might get serialized and deserialized in a database. It's sort of less worry to control all of that interaction by keeping it in a string.

Reply:
Fair enough. Thanks for considering.